tools/hpcc_aks_util.pl

```diff
@@ -0,0 +1,772 @@
+# Includes configurable Enterprise Agreement and Dds_v4 family discounts
+sub get_vm_hourly_cost {
+    my ($vm_size) = @_;
+    my $ea_multiplier = (100 - $ea_discount) / 100;
+    my $ds_v4_family_multiplier = (100 - $ds_v4_family_discount) / 100;
+    
+    my %vm_costs = (
+        # Standard D-series v5 (EA discount applied)
+        'Standard_D2s_v5' => 0.096 * $ea_multiplier,
+        'Standard_D4s_v5' => 0.192 * $ea_multiplier,
+        'Standard_D8s_v5' => 0.384 * $ea_multiplier,
+        'Standard_D16s_v5' => 0.768 * $ea_multiplier,
+        'Standard_D32s_v5' => 1.536 * $ea_multiplier,
+        'Standard_D48s_v5' => 2.304 * $ea_multiplier,
+        'Standard_D64s_v5' => 3.072 * $ea_multiplier,
+        
+        # Standard D-series v4 (smaller sizes - EA discount applied)
+        'Standard_D2s_v4' => 0.096 * $ea_multiplier,
+        'Standard_D4s_v4' => 0.192 * $ea_multiplier,
+        'Standard_D8s_v4' => 0.384 * $ea_multiplier,
+        
+        # Standard Dds_v4 family (D16ds v4 and larger, Linux) - Dds_v4 family discount applied
+        'Standard_D16s_v4' => 0.768 * $ds_v4_family_multiplier,
+        'Standard_D32s_v4' => 1.536 * $ds_v4_family_multiplier,
+        'Standard_D48s_v4' => 2.304 * $ds_v4_family_multiplier,
+        'Standard_D64s_v4' => 3.072 * $ds_v4_family_multiplier,
+        
+        # Standard E-series v5 (memory optimized - EA discount applied)
+        'Standard_E2s_v5' => 0.126 * $ea_multiplier,
+        'Standard_E4s_v5' => 0.252 * $ea_multiplier,
+        'Standard_E8s_v5' => 0.504 * $ea_multiplier,
+        'Standard_E16s_v5' => 1.008 * $ea_multiplier,
+        'Standard_E32s_v5' => 2.016 * $ea_multiplier,
+        'Standard_E48s_v5' => 3.024 * $ea_multiplier,
+        'Standard_E64s_v5' => 4.032 * $ea_multiplier,
+        
+        # Standard F-series v2 (compute optimized - EA discount applied)
+        'Standard_F2s_v2' => 0.085 * $ea_multiplier,
+        'Standard_F4s_v2' => 0.169 * $ea_multiplier,
+        'Standard_F8s_v2' => 0.338 * $ea_multiplier,
+        'Standard_F16s_v2' => 0.676 * $ea_multiplier,
+        'Standard_F32s_v2' => 1.352 * $ea_multiplier,
+        'Standard_F48s_v2' => 2.028 * $ea_multiplier,
+        'Standard_F64s_v2' => 2.704 * $ea_multiplier,
+        'Standard_F72s_v2' => 3.042 * $ea_multiplier,
+        
+        # Standard B-series (burstable - EA discount applied)
+        'Standard_B2s' => 0.041 * $ea_multiplier,
+        'Standard_B2ms' => 0.083 * $ea_multiplier,
+        'Standard_B4ms' => 0.166 * $ea_multiplier,
+        'Standard_B8ms' => 0.333 * $ea_multiplier,
+        'Standard_B12ms' => 0.499 * $ea_multiplier,
+        'Standard_B16ms' => 0.666 * $ea_multiplier,
+        'Standard_B20ms' => 0.832 * $ea_multiplier
+    );
+    
+    if (exists $vm_costs{$vm_size}) {
+        return sprintf("%.3f", $vm_costs{$vm_size});
+    } else {
+        # Default fallback - estimate based on CPU count if available
```

Comment:
if this happens, won't it be very misleading? i.e. silently defaulting figures if an unknown VM type is found.
I think it would be better to error, with the message saying the vmcosts table needs updating.